lib/xgu/xgu.h

```diff
@@ -712,6 +779,99 @@ uint32_t* xgu_set_light_local_attenuation(uint32_t* p, unsigned int light_index,
     return push_floats(p, v.f, 3);
 }
 
+/* ==== Texture stuff ==== */
+
+inline
+uint32_t* xgu_set_texture_offset(uint32_t* p, uint32_t offset) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_OFFSET, offset);
+}
+
+inline
+uint32_t* xgu_set_texture_format(uint32_t* p, uint8_t context_dma, bool cubemap_enable,
+                                 XguBorderSrc border_src, uint8_t dimensionality,
+                                 XguTexFormatColor format, uint8_t mipmap_levels,
+                                 uint8_t u_size, uint8_t v_size, uint8_t p_size) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_FORMAT,
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_CONTEXT_DMA, context_dma) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_CUBEMAP_ENABLE, cubemap_enable) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BORDER_SOURCE, border_src) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_DIMENSIONALITY, dimensionality) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_COLOR, format) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_MIPMAP_LEVELS, mipmap_levels) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_U, u_size) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_V, v_size) |
+                                  XGU_MASK(NV097_SET_TEXTURE_FORMAT_BASE_SIZE_P, p_size));
+}
+
+inline
+uint32_t* xgu_set_texture_address(uint32_t* p, uint32_t address) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_ADDRESS, address);
+}
+
+inline
+uint32_t* xgu_set_texture_control0(uint32_t* p, bool enable, uint16_t min_lod, uint16_t max_lod) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_CONTROL0,
+                                  (enable ? NV097_SET_TEXTURE_CONTROL0_ENABLE : 0) |
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL0_MIN_LOD_CLAMP, min_lod) |
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL0_MAX_LOD_CLAM, max_lod));
+}
+
+inline
+uint32_t* xgu_set_texture_control1(uint32_t* p, uint16_t pitch) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_CONTROL1,
+                                  XGU_MASK(NV097_SET_TEXTURE_CONTROL1_IMAGE_PITCH, pitch));
+}
+
+inline
+uint32_t* xgu_set_texture_filter(uint32_t* p, uint16_t lod_bias, uint8_t filter_min, uint8_t filter_mag,
+                                 bool alpha_signed, bool r_signed, bool g_signed, bool b_signed) {
```

Comment:
Signed bits could be handled as masks like we do for `xgu_set_colormask` / `xgu_clear` (or whatever else uses RGBA masks - we should make it consistent)

If we keep it like this, I prefer `a_signed` instead of `alpha_signed`... or... `red_signed` etc. instead of `r_signed`.

I also generally prefer RGBA order (but it should be consistent) as that's traditionally mapping to XYZW=RGBA in graphics.